### PR TITLE
raise error when creating second open JudgeDecisionReviewTask

### LINKED ISSUE #12802: Prevent creation of redundant JudgeDecisionReviewTasks
## Description
Occasionally when we create the JudgeDecisionReviewTask & its child AttorneyTask, we do this twice. This can lead to confusion on the part of the Attorney and/or Judge this is assigned to, and will have unknown behavior if the Judge pushes through checkout twice.

## Background/context/resources

Tomas has done some research and found that we are doing two distinct POSTs, creating the redundant task branch.  

[Investigation](https://dsva.slack.com/archives/CJL810329/p1573842570123600)
[POST discussion](https://dsva.slack.com/archives/CAM9FJ85P/p1573866025061200)

Possible places to look:
- is the front end allowing multiple form submissions?
- is the front end mistakenly sending multiple POST requests for a single form submission?
- is something in our networking setup glitching and duplicating POST requests?

## Acceptance Criteria
 - [ ] Any outstanding affected appeals are corrected in Prod
 - [ ] Root cause of the double DecisionReview branch is addressed

### PR DESCRIPTION
**Pull request by yoomlam**
Resolves #12802 and probably #12824

### Description
Prevent duplicate open JudgeDecisionReviewTask since there should only ever be one open JudgeDecisionReviewTask at a time for an appeal.

Added `verify_user_task_unique` check `before_create` of `JudgeDecisionReviewTask` instances.

Added new tests.

### Acceptance Criteria
- [ ] Code compiles correctly and tests pass

### CODE REVIEW

File: app/models/tasks/judge_decision_review_task.rb [hunk 1]
```diff
@@ -16,4 +18,23 @@ def additional_available_actions(_user)
   def self.label
     COPY::JUDGE_DECISION_REVIEW_TASK_LABEL
   end
+
+  # Use the existence of another open JudgeDecisionReviewTask to prevent duplicates since there should only
+  # ever be one open JudgeDecisionReviewTask at a time for an appeal.
+  def verify_user_task_unique
+    return if !open?
```
Comment (lomky) [inline app/models/tasks/judge_decision_review_task.rb]: is this checking if the task being created is open?

Comment (yoomlam) [reply app/models/tasks/judge_decision_review_task.rb]: Yes, copied it from https://github.com/department-of-veterans-affairs/caseflow/blob/3f8f454f0ed930ff381ebfd10eb91a6b58b089be/app/models/task.rb#L213

File: spec/models/tasks/judge_task_spec.rb [hunk 2]
```diff
@@ -124,7 +124,51 @@
...
+      let!(:jdrt) do
+        create(:ama_judge_decision_review_task, appeal: root_task.appeal, parent: root_task, assigned_to: judge)
+      end
+
+      subject { JudgeDecisionReviewTask.create!(appeal: root_task.appeal, parent: root_task, assigned_to: judge) }
+
+      context "when an open (assigned) JudgeDecisionReviewTask already exists" do
+        it "should fail creation of second JudgeDecisionReviewTask" do
+          expect { subject }.to raise_error(Caseflow::Error::DuplicateUserTask)
+          expect(root_task.appeal.tasks.count).to eq(2), root_task.appeal.tasks.to_a.to_s
```
Comment (lomky) [inline spec/models/tasks/judge_task_spec.rb]: nit pick: I think I'd prefer we stored how many tasks exist on the appeal in a variable before calling `subject`, and then check it hasn't changed after? That feels more robust if we ever change the factory, and gives a meaningful name over just a number. (ditto across the rest of the tasks). I really do like being robust to check the task tree hasn't changed!

Comment (yoomlam) [reply spec/models/tasks/judge_task_spec.rb]: Done

File: spec/models/tasks/judge_task_spec.rb [hunk 3]
```diff
@@ -127,42 +127,32 @@
...
-          expect(root_task.appeal.tasks.count).to eq(2), root_task.appeal.tasks.to_a.to_s
-        end
-      end
-
-      context "when an open (on_hold) JudgeDecisionReviewTask already exists" do
-        it "should fail creation of second JudgeDecisionReviewTask" do
-          jdrt.update(status: Constants.TASK_STATUSES.on_hold)
-          expect { subject }.to raise_error(Caseflow::Error::DuplicateUserTask)
-          expect(root_task.appeal.tasks.count).to eq(2), root_task.appeal.tasks.to_a.to_s
+      Task.open_statuses.each do |o_status|
```
Comment (lomky) [inline spec/models/tasks/judge_task_spec.rb]: Beautiful! :tada:

**Pull request merged**